rfd/0180-sso-as-mfa-method.md

```diff
@@ -20,31 +21,152 @@ moderated sessions. Ideally we will support SSO as a first class MFA method.
 
 ## Why
 
-Delegating MFA checks to a registered IdP has several benefits:
+Delegating MFA checks to a registered IdP has several potential benefits:
 - Administrators gain the ability to configure and monitor all authentication
 directly through an IdP.
-- Teleport can integrate with custom MFA mechanisms and devices through an IdP.
-- Improves UX for SSO users without an MFA device registered.
-- Allows SSO users to add their first MFA device securely.
+- Teleport can integrate with custom MFA mechanisms and devices registered
+setup through an IdP.
+- Improves UX for new SSO users by providing an MFA flow without requiring the
+user to manually register a new device.
+- Improve UX for existing users by providing multiple paths during MFA checks,
+such as when they don't have their WebAuthn device plugged in.
+- Improve security of new SSO users by requiring an SSO MFA check to add their
+first MFA device.
 
 ## Details
 
-### UX and User stories
+### UX
+
+This feature will provide users with additional paths towards passing MFA
+checks. In some cases, such as when the user has no MFA device registered,
+this will improve UX.
 
-This feature will improve the UX of performing MFA checks for SSO users by
-removing the requirement to add an MFA device. The primary UX concerns for
-this feature are:
-- Adding too many options without clarity (OTP, Webauthn, and SSO).
+Adding an SSO MFA option also comes with potential downsides:
+- Providing too many options without clarity (OTP, Webauthn, and SSO).
 - Locking users into SSO MFA in cases where WebAuthn may be preferred.
 - Automatically opening browser windows when WebAuthn may be preferred.
 
+These concerns will be addressed in the sections below.
+
+#### Devices
+
+When an auth connector has MFA enabled, Teleport will treat it as if any user
+originating from that auth connector has a corresponding SSO MFA device
+registered. This way, users can view their true MFA options.
+
+```console
+> tsh mfa ls
+Name     Type     Added at Last used                     
+-------- -------- -------- ---------
+yubi     WebAuthn ...      ...
+okta     SAML     ...      ...
```

Comment:
Re last used: seems alright.

Re added at: it could be the create time of the user. I wouldn't do anything too complex for this, neither of these are too important.

> Between the type and the name, naming conflicts shouldn't be possible

Right, SSOMFADevice.  👍